app/assets/javascripts/modules/SearchFilter.js

```diff
@@ -0,0 +1,31 @@
+define(['jquery'], function($) {
+
+  'use strict';
+
+  var $element = $('[data-search-filter]'),
+      $trigger = $('[data-search-filter-trigger]'),
+      $triggerIcon = $('[data-search-filter-icon]'),
+      $triggerHeading = $('[data-search-heading-second]'),
+      $target = $('[data-search-filter-target]');
+
+  $element.each(function() {
+    var $this = $(this);
```

Comment:
You won't event need the `$.each` as jQuery will act upon an element collection when using `.on`.